src/algebra/char_p/basic.lean

```diff
@@ -338,7 +338,8 @@ lemma cast_eq_mod (p : ℕ) [char_p R p] (k : ℕ) : (k : R) = (k % p : ℕ) :=
 calc (k : R) = ↑(k % p + p * (k / p)) : by rw [nat.mod_add_div]
          ... = ↑(k % p)               : by simp [cast_eq_zero]
 
-theorem char_ne_zero_of_fintype (p : ℕ) [hc : char_p R p] [fintype R] : p ≠ 0 :=
+theorem char_ne_zero_of_fintype (R : Type*) [non_assoc_ring R] [fintype R]
```

Comment:
Instead of adding `non_assoc_ring` here, you can just change `ring` to `non_assoc_ring` on line 330

Reply:
OK.